contracts/FastExitHandler.sol

```diff
@@ -103,19 +103,21 @@ contract FastExitHandler is ExitHandler {
   }
 
   function unpackSignedData(
-    bytes32[] signedData
+    bytes32[] memory signedData
   ) internal pure returns (
     uint256 buyPrice, bytes32 utxoId, address signer
   ) {
-    bytes32[] memory sigBuff = new bytes32[](2);
     utxoId = signedData[0];
     buyPrice = uint256(signedData[1]);
     bytes32 r = signedData[2];
     bytes32 s = signedData[3];
-    uint8 v = uint8(signedData[4]);
-    sigBuff[0] = utxoId;
-    sigBuff[1] = signedData[1];
-    bytes32 sigHash = keccak256(sigBuff);
+    uint8 v = uint8(uint256(signedData[4]));
+    bytes32 sigHash = signedData[1];
+    assembly {
+      mstore(0, utxoId)
+      mstore(0x20, sigHash)
+      sigHash := keccak256(0, 0x40)
```

Comment:
why this change?

Reply:
because it would have required `abi.encoded`, which is not gas efficient.

Reply:
https://twitter.com/nicksdjohnson/status/1044904764146962432